webapp/certification/views.py

```diff
@@ -71,13 +86,96 @@ def certification_home():
             else:
                 server_releases[release] = vendor[release]
 
-    return flask.render_template(
-        "certification/index.html",
-        desktop_releases=desktop_releases,
-        desktop_vendors=desktop_vendors,
-        server_releases=server_releases,
-        iot_releases=iot_releases,
-        iot_vendors=iot_vendors,
-        soc_releases=soc_releases,
-        soc_vendors=soc_vendors,
-    )
+    if request.args:
+        query = request.args.get("text", default=None, type=str)
+        limit = request.args.get("limit", default=20, type=int)
+        offset = request.args.get("offset", default=0, type=int)
+
+        forms = (
+            ",".join(request.args.getlist("form"))
+            if request.args.getlist("form")
+            else None
+        )
+        releases = (
+            ",".join(request.args.getlist("release"))
+            if request.args.getlist("release")
+            else None
+        )
+        vendors = (
+            ",".join(request.args.getlist("vendor"))
+            if request.args.getlist("vendor")
+            else None
+        )
+
+        models_response = api.certified_models(
+            category__in=forms,
+            major_release__in=releases,
+            vendor=vendors,
+            query=query,
+            offset=offset,
+        )
+
+        # Filters and result numbers
+        form_filters = {
+            "Laptop": 0,
+            "Desktop": 0,
+            "Ubuntu Core": 0,
+            "Server": 0,
+            "Server SoC": 0,
+        }
+        release_filters = {}
+        for release in all_releases:
+            release_filters[release] = 0
+
+        vendor_filters = {}
+        for vendor in all_vendors:
+            vendor_filters[vendor] = 0
+
+        results = models_response["objects"]
+
+        # Populate filter numbers
+        for index, model in enumerate(results):
```

Comment:
I'm not entirely sure I get this, but if what we want to do here is to get the number of results per filter type should we not be incrementing by one instead?
```
form_filters[model["category"]] += 1
release_filters[model["release"]] += 1
vendor_filters[model["make"]] += 1
```

Reply:
Done.